### PR TITLE
feat: Adding FST architecture diagrams & documentation

### LINKED ISSUE #395: feat: Add FST architecture documentation with diagrams

### PR DESCRIPTION
**Pull request by deepak-swirlds**
## Description

This pull request changes the following:

- Adds documentation under docs/ folder

### Related Issues

- Closes #395

### CODE REVIEW

File: docs/Terminology.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - Ingress Operator
+    - Sealed secrets operator ?
+
+Settings and Configuration
+- Platform Settings
+    - setting.txt
+- Platform Configuration
+    - config.txt
+- Node Software Address Books
+    - account  101 .bin
```
Comment (jeromy-cannon) [inline docs/Terminology.md:52]:
```suggestion
    - account 101 .bin
```

File: docs/architecture/flow diagram-high level.drawio.svg
Comment (jeromy-cannon) [inline docs/architecture/flow diagram-high level.drawio.svg:1]: what is the reason why 'get IPs of statefulset(s)' is listed twice?

Comment (jeromy-cannon) [inline docs/architecture/flow diagram-high level.drawio.svg:1]: what is the reason why '[all pods] copy hedera-build.zip' is listed twice?

**Review by leninmehedy — COMMENTED** [review]
Let's keep design docs in Notion until it is finalized. I don't see the need of merging it and store in the repo yet.

When we are merging interface definitions after a design sprint/round, we need to ensure proper doc blocks are added so that we can auto-generate proper java doc.

**Comment by leninmehedy** [comment]
@deepak-swirlds Let's put the design docs in Notion [here](https://www.notion.so/swirldslabs/Solution-Design-Options-882f0804efd24f67b82834b6375ac680?pvs=4) and close this PR.

**Comment by leninmehedy** [comment]
Closing it as discussed

**Pull request closed without merging**